examples/irsa_autoscale_refresh/main.tf

```diff
@@ -0,0 +1,211 @@
+        }
+        triggers = ["tag"]
+      }
+
+      propogate_tags = [{
+        key                 = "aws-node-termination-handler/managed"
+        value               = true
+        propagate_at_launch = true
+      }]
+    }
+
+    mixed_instance = {
+      create_launch_template = true
+      launch_template_name   = "mixed-instance"
+
+      use_mixed_instances_policy = true
+      mixed_instances_policy = {
+        instances_distribution = {
+          on_demand_base_capacity                  = 0
+          on_demand_percentage_above_base_capacity = 10
+          spot_allocation_strategy                 = "capacity-optimized"
+        }
+
+        override = [
+          {
+            instance_type     = "m5.large"
+            weighted_capacity = "1"
+          },
+          {
+            instance_type     = "m6i.large"
+            weighted_capacity = "2"
+          },
+        ]
+      }
+
+      propogate_tags = [{
+        key                 = "aws-node-termination-handler/managed"
+        value               = true
+        propagate_at_launch = true
+      }]
+    }
+
+    spot = {
+      create_launch_template = true
+      launch_template_name   = "spot"
+
+      instance_type = "m5.large"
+      instance_market_options = {
+        market_type = "spot"
+      }
+
+      bootstrap_extra_args = "--kubelet-extra-args '--node-labels=node.kubernetes.io/lifecycle=spot'"
+
+      propogate_tags = [{
+        key                 = "aws-node-termination-handler/managed"
+        value               = true
+        propagate_at_launch = true
+      }]
+    }
+  }
+
+  tags = merge(local.tags, { Foo = "bar" })
+}
+
+################################################################################
+# aws-auth configmap
+# Only EKS managed node groups automatically add roles to aws-auth configmap
+# so we need to ensure fargate profiles and self-managed node roles are added
+################################################################################
+
+data "aws_eks_cluster_auth" "this" {
+  name = module.eks.cluster_id
+}
+
+locals {
+  kubeconfig = yamlencode({
+    apiVersion      = "v1"
+    kind            = "Config"
+    current-context = "terraform"
+    clusters = [{
+      name = module.eks.cluster_id
+      cluster = {
+        certificate-authority-data = module.eks.cluster_certificate_authority_data
+        server                     = module.eks.cluster_endpoint
+      }
+    }]
+    contexts = [{
+      name = "terraform"
+      context = {
+        cluster = module.eks.cluster_id
+        user    = "terraform"
+      }
+    }]
+    users = [{
+      name = "terraform"
+      user = {
+        token = data.aws_eks_cluster_auth.this.token
+      }
+    }]
+  })
+}
+
+resource "null_resource" "apply" {
+  triggers = {
+    kubeconfig = base64encode(local.kubeconfig)
+    cmd_patch  = <<-EOT
+      kubectl create configmap aws-auth -n kube-system --kubeconfig <(echo $KUBECONFIG | base64 --decode)
```

Comment:
this can be very problematic in case of terraform cloud probably

Reply:
agreed! this is just a quick and dirty way to facilitate testing and demonstrate steps users may need to take/handle on their end. what this change is proposing here is that managing/updating/creating the aws-auth configmap is now up to users and outside the scope of this module. you can do hacky kubectl way like shown (least amount of effort, again just for testing and verification), you can integrate it with your gitops approach if so desired, etc. There are numerous ways, but it greatly simplifies the module by not prescribing and managing the aws-auth configmap going forward (IMO)